### PR TITLE
Implementing getAttributeNode() and similar methods.

### PR DESCRIPTION
**Pull request by martiansideofthemoon**
Attempting to solve #8066 Does it look good so far @Manishearth ?

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/8068)

### CODE REVIEW

**Comment by nox** [comment]
These should return attributes (`Option<Root<Attr>>`), not their values (`Option<DOMString>`).

File: components/script/dom/webidls/Element.webidl [hunk 1]
```diff
@@ -37,6 +37,10 @@ interface Element : Node {
   DOMString? getAttribute(DOMString name);
   [Pure]
   DOMString? getAttributeNS(DOMString? namespace, DOMString localName);
+  [Pure]
+  DOMString? getAttributeNode(DOMString name);
```
Comment (Manishearth) [inline components/script/dom/webidls/Element.webidl]: This is incorrect, see https://dom.spec.whatwg.org/#node

**Comment by martiansideofthemoon** [comment]
@Manishearth Hope it is better now

**Comment by Manishearth** [comment]
Looks good to me. You need to update the test expectations like you did in #8038

**Comment by martiansideofthemoon** [comment]
@Manishearth Well I am having issues with changing these `.ini` files. I always get permission issues unless I run `sudo su` before editing them. Also, the automatic method [here] (http://mxr.mozilla.org/servo/source/tests/wpt/README.md#80) doesn't seem to help. It took an hour to execute and I get a `Permission Denied` for every test file on running `git diff` or `git commit`.

**Comment by Manishearth** [comment]
You seem to have done the checkout under the wrong user, perhaps?

**Comment by Manishearth** [comment]
(Also, the r=manishearth isn't necessary)

**Comment by martiansideofthemoon** [comment]
@Manishearth sorry, didn't get what you meant by "checkout under the wrong user".
I had run the ./mach command to run wpt tests using `sudo`, if that makes any difference.

**Comment by Manishearth** [comment]
yeah; you shouldn't have done that :smile: . Servo doesn't need sudo aside from `apt-get`ing dependencies

Try `sudo chown -R <username> tests/wpt`.

**Comment by Ms2ger** [comment]
```

/dom/interfaces.html
--------------------
PASS expected FAIL Element interface: operation getAttributeNode(DOMString)
PASS expected FAIL Element interface: operation getAttributeNodeNS(DOMString,DOMString)
```

You'll need to update those expectations; see `tests/wpt/README.md`.

**Comment by Manishearth** [comment]
You can use the log from http://build.servo.org/builders/linux-rel/builds/569/steps/shell_1/logs/test-wpt.log to update.

**Comment by martiansideofthemoon** [comment]
@Manishearth `chown` seems to have fixed all issues :smile: Removed two additional test expectations.

**Comment by martiansideofthemoon** [comment]
@Manishearth What instructions am I allowed to give @bors-servo ?

**Comment by Manishearth** [comment]
None of them :smile: 

bors only listens to reviewers. This might change in the future.

**Comment by martiansideofthemoon** [comment]
Resolves #8066

**Pull request merged**